hazelcast-spring/src/main/java/com/hazelcast/spring/cache/HazelcastCache.java

```diff
@@ -136,6 +145,14 @@ public ValueWrapper putIfAbsent(Object key, Object value) {
     }
 
     private Object lookup(Object key) {
+        if (readTimeout > 0) {
+            try {
+                return this.map.getAsync(key).get(readTimeout, TimeUnit.MILLISECONDS);
+            } catch (Exception ex) {
```

Comment:
we should not catch the general Exception. I think we need to call some time-out related exception + perhaps the `ExecutionException` and `unpeel()` it.